_posts/2017/2017-09-01-socrates-2017-order-is-optional.md

```diff
@@ -0,0 +1,97 @@
+* If you see the candidate struggles, offer to lower bar, e.g. by applying junior instead intermediate level criteria.
+* After the interview have a retro with the candidate, ask for feedback, and give feedback!
+You might want to offer a private, non-company-related communication channel, too.
+
+### The Medium Interview Handbook
+
+In the second session on Saturday we dissected [Medium's Interview Handbook](https://medium.engineering/mediums-engineering-interview-process-b8d6b67927c4) with a focus on the [grading criteria](https://medium.engineering/engineering-interviews-grading-rubric-8b409bec021f).
+While we found some weak spots of this manual, it is still an incredible resource that helps you assess a candidate's fit.
+
+### Oh, it's about software development and technology, too
+
+A SoCraTes is always a great place to level up your TDD, design, and mob programming skills practicing katas.
+Playing the [Mob Programming RPG by Willem Larsen](https://github.com/willemlarsen/mobprogrammingrpg/) with a fizzbuzz kata we learned good mob programming behavior, and pairing practices.
+For maximum effect, make some sound like "KAAASHINGG" when leveling up as an indicator to the group, that your behavior was beneficial.
+Laughs included.
+
+Obviously, there were Vim vs. Emacs bashes (or [fish](https://fishshell.com/)es?).
+Vim clearly won this time around.
+I was able to share knowledge I gained at university in a session about compilers, explaining the basic steps of language processing, grammars, and syntax trees.
+
+Also a talk about the BEAM virtual machine (VM), the Erlang/Elixir/LFE runtime, drew my attention.
+Actually the BEAM VM is more like an operating system than a runtime: It prioritizes processes (Erlang functions), and assigns them to schedulers.
+Inter-process communication happens via messages in [Erlang](https://www.erlang.org/).
+Those are copied from the source to the target process, no memory is shared.
+The VM's failure recovery mechanism can be used to strictly separate business logic from exceptional state handling as well as to control program flow in a distributed computing environment.
+Linked processes get killed, if just one of the linked processes dies (this also applies transitively).
+Supervisors can then restart killed processes, if needed, thus gracefully handling, and recovering non-happy-path executions.
+
+One gem was saved for last: On Sunday [Xavier Detant](https://twitter.com/xdetant?lang=en) gave an excellent [introduction to functional programming](https://github.com/FaustXVI/functional-programming-introduction/tree/playing), transforming an imperative program to a functional one, step by step, eventually deriving a valid Haskell program from the original JavaScript source code ("Haskell is just some syntactic sugar for JavaScript").
+If you want to replay the steps taken during the talk, switch to the `playing` branch and check the commits.
+
+## Conclusion
```

Comment:
`Reduction` was a reference to the functional programming talk mentioned before.

I could also live with:
```
## Reduce(notes)(essence)
```
... making the functional reference more explicit. Conclusion is fine too, just a bit boring :)